python/desc/skycatalogs/utils/sed_tools.py

```diff
@@ -0,0 +1,230 @@
+            my_Llambda.extend((Llambda[i], Llambda[i]))
+
+        # Convert to (unredshifted) flux given redshift_hubble.
+        flambda = np.array(my_Llambda)/(4.0*np.pi*self.dl(redshift_hubble)**2)
+
+        # Convert to cgs units
+        flambda *= (1e7/1e4)  # (erg/joule)*(m**2/cm**2)
+
+        # Create the lookup table.
+        lut = galsim.LookupTable(self.wl_deltas, flambda, interpolant='nearest')
+
+        if resolution:
+            wl_min = min(self.wl_deltas)
+            wl_max = max(self.wl_deltas)
+            wl_res = np.linspace(wl_min, wl_max, int((wl_max - wl_min)/resolution))
+            flambda_res = [lut(wl) for wl in wl_res]
+            lut = galsim.LookupTable(wl_res, flambda_res, interpolant='linear')
+
+        # Create the SED object and apply redshift.
+        sed = galsim.SED(lut, wave_type='nm', flux_type='flambda')\
+                    .atRedshift(redshift)
+
+        return sed
+
+    def create_pointsource(self, rel_path, redshift=0):
+        '''
+        Return a galsim SED from information in a file
+        '''
+        fpath = os.path.join(self.sims_sed_library_dir, rel_path)
+
+        sed = galsim.SED(fpath, wave_type='nm', flux_type='flambda')
+        if redshift > 0:
+            sed = sed.atRedshift(redshift)
+        return sed
+
+    def magnorm(self, tophat_values, z_H):
+        one_Jy = 1e-26  # W/Hz/m**2
+        Lnu = tophat_values[self.ix_500nm]*self._to_W_per_Hz  # convert to W/Hz
+        Fnu = Lnu/4/np.pi/self.dl(z_H)**2
+        with np.errstate(divide='ignore', invalid='ignore'):
+            return -2.5*np.log10(Fnu/one_Jy) + 8.90
+
+class Extinguisher:
+    '''
+    Applies extinction to a SED
+    '''
+    def __init__(self, sed_factory, ext_bin_width=0.1, mwRv=3.1):
+        '''
+        sed_factory         instance of ObservedSedFactory
+
+        '''
+
+        # Save bins to be used for extinction. F19x_range is in units of
+        # 1/micron so convert.
+        wl_ext_min = 1e3/F19.x_range[1]
+        wl_ext_max = 1e3/F19.x_range[0]
+        wl_extinct = np.linspace(wl_ext_min, wl_ext_max,
+                                 int((wl_ext_max - wl_ext_min)/ext_bin_width))
+        self.wl_extinct = wl_extinct
+        self.wl_extinct_nm = wl_extinct*u.nm
+        self.extinction = F19(Rv=mwRv)
+
+        my_wl = sed_factory.wl_deltas
+        my_wl = my_wl[np.where((wl_ext_min < my_wl) & (my_wl < wl_ext_max))]
+        self.wl_deltas = my_wl
+        self.wl_deltas_u = my_wl*u.nm
+
+    def extinguish(self, sed, mwAv):
+        ext = self.extinction.extinguish(self.wl_deltas_u, Av=mwAv)
+        spec = galsim.LookupTable(self.wl_deltas, ext, interpolant='linear')
+        mw_ext = galsim.SED(spec, wave_type='nm', flux_type='1')
+        sed = sed*mw_ext
+
+        return sed
+
+class AB_mag:
+    """
+    Convert flux to AB magnitude for a set of bandpasses.
+    """
+    def __init__(self, bps):
+        ab_sed = galsim.SED(lambda nu : 3631e-23, wave_type='nm',
```

Comment:
This is actually an approximation to the exact definition (see https://en.wikipedia.org/wiki/AB_magnitude).  Using the definition, this should be
```
ab_sed = galsim.SED(lambda nu : 10**(8.90/2.5 - 23), wave_type='nm', flux_type='fnu')
```

Reply:
Fixed in next push